### ISSUE TITLE
Muon Upgrade: Pinning to Recurring Tips workflow

### ISSUE THREAD

**Issue by mandar-brave** [brave/brave-browser#1910]
Issue:
For users that will be silently upgraded from Muon to Brave Core, and ones who have pinned certain sites for fixed donations, the upgrade needs to handle

a) migrating those sites from pins to recurring tips
b) compute their BAT montlhy based on Muon monthly budget and % of pinning amount and round it up to the nearest integer
c) let the user know when they get to recurring tips that they can go to settings to remove sites if they choose to discontinue monthly tips (see image)

Image for when user has been upgraded from Muon and has pinned sites that are migrated to Recurring Tips:

![image](https://user-images.githubusercontent.com/35932307/47767095-4de3aa00-dc8f-11e8-91af-eeae44f9d5f6.png)

On clicking Review List on the image, the settings panel for the user's ability to add/delete sites should be displayed.

cc @davidtemkin

**Comment by jenn-rhim** [comment]
Here is the results of the discussion @davidtemkin @mandar-brave @mrose17 @NejcZdovc  @bradleyrichter 

1. Existing Brave Payments user comes to BR settings page
2. Pinned site conversion to monthly tips has already occurred, Tip window is already populated with imported pin(s).
3. Auto Contribute may or may not be enabled depending on result of pinned site import.  **If any monthly budget remains following deduction of pinned allocation, the remaining BAT is rounded to the nearest option on the monthly auto-contribute payment menu and set as the monthly payment. If the remaining BAT following deduction of pinned allocation is lower than the smallest option on the menu, Auto Contribute is turned off.** 
4. A footer at the bottom of the Tip window tells them that their pinned sites have been converted into monthly tips. It also has a “learn more” button”.
5. If the user presses “learn more”, an explanatory dialog comes up telling them how it works. It can only dismissed with “OK”.
6. That footer persists until either the user views it or until seven days have passed.

For existing Brave Payments users who have pinned sites, the initial state of the Tips panel  below **will not be shown**.
![screen shot 2018-11-01 at 5 55 13 pm](https://user-images.githubusercontent.com/4369856/47887991-56f38900-ddff-11e8-9222-4ffbb6aa30b3.png)


Instead, the panel will pre-populate the pinned sites as 'sites with recurring monthly tip'.
![screen shot 2018-11-01 at 5 12 51 pm](https://user-images.githubusercontent.com/4369856/47888049-9ae68e00-ddff-11e8-8885-533e04eeec81.png)
*This bottom bar with '**Learn More**' that takes the user to the dialog will stay until the user clicks on it or x days. 


**Learn More** takes the user to the information dialog:
![screen shot 2018-11-01 at 1 46 15 pm](https://user-images.githubusercontent.com/4369856/47888172-1d6f4d80-de00-11e8-9434-8ea336d5d919.png)

**Comment by kjozwiak** [comment]
> For existing Brave Payments users who have pinned sites, the initial state of the Tips panel  below **will not be shown**.
> ![screen shot 2018-11-01 at 5 55 13 pm](https://user-images.githubusercontent.com/4369856/47887991-56f38900-ddff-11e8-9222-4ffbb6aa30b3.png)

Should we be mentioning tips via "Like" buttons even though it won't be implemented by the time we start migrating users? It might cause confusion as users might expect this to work but in reality, it wasn't implemented/added into the product.

**Comment by davidtemkin** [comment]
@kjozwiak can you file that (tipping with "like") as its own issue? It is a problem

**Comment by NejcZdovc** [comment]
@kjozwiak designs for off/disabled state were already implemented

**Comment by mandar-brave** [comment]
For computing the tips @bsclifton 
To clarify, for each pinned value, we compute

pinnedPercentage * monthlyBudget

then we round the result downward to the nearest lower int (floor())
that, if non-zero, becomes the monthly tip amount for that pinned publisher
if zero, remove the pin

**Comment by davidtemkin** [comment]
pasting from Slack -- **note how this affects auto-contribute**

If we always round down to the nearest (lower) integer value we can never exceed monthly budget. Some things get dropped because they go to zero.

Then , we look at the total that has been allocated to pins. Let’s say that you had a monthly budget of 25 BAT and 15 are allocated to pins. We take the 15 and follow this logic:

— If it’s lower than the lowest option on the monthly payment menu, we set auto-contribute to “off” — that is to say that there is no attention-based monthly contribution
— Otherwise we round the non-pinned (remaining) downward to the nearest lower integer and make that the monthly budget. (If we needed to , we could set it to a value that is on the menu, but I don’t believe that’s necessary).
To clarify, for each pinned value, we compute

pinnedPercentage * monthlyBudget

then we round the result downward to the nearest lower int (floor())
that, if non-zero, becomes the monthly tip amount for that pinned publisher
Then, having subtracted the total value of each monthly tip (that is , the sum of the downward-rounded integers) we calculate the new monthly payment (new terminology) for auto contribute , if any , per above.
newMonthlyPayment = floor(oldMonthlyBudget - sumOfMonthlyTips)
if (newMonthlyPayment < lowestOptionOnMonthlyPaymentMenu)
   set auto-contribute to off:
* getting list of options on menu requires a server call
